src/tcp/user_buffer.ml

```diff
@@ -59,6 +59,9 @@ module Rx = struct
     | None -> 0
     | Some b -> Cstruct.length b
 
+  let add_l t s =
+    ignore(Lwt_dllist.add_l (Some s) t.q)
```

Comment:
any idea whether any other things must be updated? I frankly don't understand much of the `add_r` below, but it deals with various `cur_size` and `max_size`.

Reply:
`cur_size` and `max_size` seem to be r(elated to a window of available data in the buffer (`cur_size` is what is currently available and `max_size` the bound for available data, but it should be possible to exceed that limit with the linked list data structure).
To keep things going, I think it's best to update `cur_size` and call `notify_size_watcher` to say that the data is online. Something like (the first comparison in `add_r` seems to be there to avoid exceeding `max_size` (again), I'm not sure the problem could be anything other than higher memory consumption, but it may be best to take care of that?):
```
  let add_l t s =
     match Lwt_dllist.take_opt_l t.readers with
     | None ->
         t.cur_size <- Int32.(add t.cur_size (of_int (seglen s)));
         ignore(Lwt_dllist.add_l (Some s) t.q)
         notify_size_watcher t 
     | Some w ->
         Lwt.return (Lwt.wakeup u s)
```